GUI/GUIWindow.cpp

```diff
@@ -2674,3 +2663,136 @@ void UI_Create() {
 
     LoadPartyBuffIcons();
 }
+
+
+String NameAndTitle(const char* name, const char* title) {
+    return localization->FormatString(
+        LSTR_FMT_S_THE_S,
+        name,
+        title
+    );
+}
+
+
+String NameAndTitle(const char* name, PLAYER_CLASS_TYPE class_type) {
+    return NameAndTitle(
+        name,
+        localization->GetClassName(class_type)
+    );
+}
+
+
+String NameAndTitle(const char* name, NPCProf profession) {
+    return NameAndTitle(
+        name,
+        localization->GetNpcProfessionName(profession)
+    );
+}
+
+
+String NameAndTitle(NPCData* npc) {
+    if (npc->pName) {
+        if (npc->profession) {
+            Assert(npc->profession < 59);
+            return NameAndTitle(npc->pName, npc->profession);
+        }
+
+        return npc->pName;
+    }
+
+    return String("");
+}
+
+
+String GetDisplayName(Actor* actor) {
+    if (actor->dword_000334_unique_name)
+        return pMonsterStats->pPlaceStrings[actor->dword_000334_unique_name];
+    else
+        return pMonsterStats->pInfos[actor->pMonsterInfo.uID].pName;
+}
+
+
+static String SeekKnowledgeElswhereString(Player *player) {
+    return localization->FormatString(
+        LSTR_FMT_SEEK_KNOWLEDGE_ELSEWHERE,
+        player->pName,
+        localization->GetClassName(player->classType)
+    )
+    + "\n \n"
+    + localization->GetString(LSTR_NO_FURTHER_OFFERS);
+}
+
+void SeekKnowledgeElswhereDialogueOption(GUIWindow* dialogue, Player* player) {
+    String str = SeekKnowledgeElswhereString(pPlayers[uActiveCharacter]);
+    int text_height = pFontArrus->CalcTextHeight(str, dialogue->uFrameWidth, 0);
+
+    dialogue->DrawTitleText(
+        pFontArrus,
+        0,
+        (174 - text_height) / 2 + 138,
+        Color16(0xFFu, 0xFFu, 0x9Bu),
+        str,
+        3
+    );
+}
+
+
+void SkillTrainingDialogue(
+    GUIWindow *dialogue,
+    int num_skills_avaiable,
+    int all_text_height,
+    int skill_price
+) {
+    if (!num_skills_avaiable) {
+        SeekKnowledgeElswhereDialogueOption(
+            dialogue,
+            pPlayers[uActiveCharacter]
+        );
+
+        return;
+    }
+
+    auto skill_price_label = localization->FormatString(
+        LSTR_FMT_SKILL_COST_D, skill_price
+    );
+    dialogue->DrawTitleText(pFontArrus, 0, 146, 0, skill_price_label, 3);
+
+    int textspacings = (149 - all_text_height) / num_skills_avaiable;
+    if (textspacings > 32) textspacings = 32;
+    int textoffset = (149 - num_skills_avaiable * textspacings - all_text_height) / 2
+            - textspacings / 2 + 162;
+    for (int i = pDialogueWindow->pStartingPosActiveItem;
+        i < pDialogueWindow->pStartingPosActiveItem + pDialogueWindow->pNumPresenceButton;
+        ++i
+    ) {
+        auto pButton = pDialogueWindow->GetControl(i);
+        int skill_id = pButton->msg_param - DIALOGUE_LEARN_STAFF;
```

Comment:
This needs further testing, crashes here now when skill_id < 0

Reply:
ah yes you're right - well that's proper annoying
should be fixed now